client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -46,58 +32,45 @@
  * interface.
  */
 public final class RestProxy implements InvocationHandler {
-    private final String host;
     private final HttpClient httpClient;
     private final SerializerAdapter<?> serializer;
-    private final SwaggerInterfaceProxyDetails interfaceDetails;
+    private final SwaggerInterfaceParser interfaceParser;
 
-    private RestProxy(String host, HttpClient httpClient, SerializerAdapter<?> serializer, SwaggerInterfaceProxyDetails interfaceDetails) {
-        this.host = host;
+    private RestProxy(HttpClient httpClient, SerializerAdapter<?> serializer, SwaggerInterfaceParser interfaceParser) {
         this.httpClient = httpClient;
         this.serializer = serializer;
-        this.interfaceDetails = interfaceDetails;
+        this.interfaceParser = interfaceParser;
     }
 
     @Override
     public Object invoke(Object proxy, final Method method, Object[] args) throws Throwable {
-        final String methodName = method.getName();
-        final SwaggerMethodProxyDetails methodDetails = interfaceDetails.getMethodProxyDetails(methodName);
-
-        final String actualHost = methodDetails.applyHostSubstitutions(host, args);
-        final String[] hostParts = actualHost.split("://");
-        final String actualPath = methodDetails.getSubstitutedPath(args);
+        final SwaggerMethodParser methodParser = interfaceParser.methodParser(method);
```

Comment:
No we don't want lazy parsing - calling the method should be fast. We should understand everything about the interface when the proxy is built and cache as much as possible.